### ISSUE TITLE
What if two methods in my app match the same event?

### ISSUE THREAD

**Issue by tombentley** [quarkiverse/quarkus-github-app#297]
The [docs](https://quarkiverse.github.io/quarkiverse-docs/quarkus-github-app/dev/developer-reference.html) describe how events result in method calls on my class, but they don't explain what happens if you have two annotated handler methods for events which match the same event.

1. Do they both get called?
2. Is there a way I can control which will get called first?

**Comment by gsmet** [comment]
Hi Tom,

Yes, they both get called. And no you can't control the order, it's CDI events being triggered. If you need ordering, you need to implement the logic in one method (and call 2 private methods if you want to share the code with others somehow).

```
The order in which observer methods are called [after firing an event] is not defined, and so portable applications should not rely upon the order in which observers are called.
```

**Comment by tombentley** [comment]
I'm not a CDI expert. Many thanks for the quick answer!

**Comment by gsmet** [comment]
Yeah no worries, I was just explaining how it was coded underneath. Let's keep this issue open as I will add a note in the doc!

**Comment by gsmet** [comment]
I added something to the doc here: https://github.com/quarkiverse/quarkus-github-app/pull/302 .

Thanks for raising the issue.
